test-cluster.sh

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+echo ECS_CLUSTER=${cluster_name} >> /etc/ecs/ecs.config
+EOF
+
+cat <<EOF > tast_def.json
+{
+    "family": "$task_family",
+    "executionRoleArn": "arn:aws:iam::782340374253:role/ecsTaskExecutionRole",
+    "networkMode": "bridge",
+    "memory": "512",
+    "containerDefinitions": [
+        {
+            "name": "dweb",
+            "image": "kontainapp/runenv-dweb",
+            "portMappings": [
+                {
+                    "containerPort": 80,
+                    "hostPort": 80,
+                    "protocol": "tcp"
+                }
+            ],
+            "essential": true,
+            "command": [
+                "./dweb",
+                "80"
+            ],
+            "interactive": true,
+            "pseudoTerminal": true
+        }
+    ],
+    "requiresCompatibilities": [
+        "EC2"
+    ],
+    "runtimePlatform": {
+        "cpuArchitecture": "X86_64",
+        "operatingSystemFamily": "LINUX"
+    }
+}
+EOF
+
+echo creating cluster
+aws --region ${region}  ecs create-cluster --cluster-name ${cluster_name} > /dev/null
+
+echo launching an Instance with the Amazon ECS AMI
+INSTANCE_ID=$(aws --region ${region} ec2 run-instances --image-id ${ami_id} --count 1 \
+    --instance-type t2.micro --key-name ezleka --tag-specifications 'ResourceType=instance,Tags=[{Key=Name,Value=KONTAIN_ECS-TEST}]' \
+    --security-group-ids ${SECURITY_GROUP_ID} \
+    --iam-instance-profile Name=ecsInstanceRoleEC2 \
+    --user-data file://instance_user_data.txt  |jq -r '.Instances | .[0] | .InstanceId')
+echo INSTANCE_ID = ${INSTANCE_ID}
+
+echo waiting for instance to reach runnign state 
+aws --region ${region} ec2 wait instance-running --instance-ids ${INSTANCE_ID} > /dev/null
+
+echo getting instance public IP
+INSTANCE_PUBLIC_IP=$(aws --region ${region} ec2 describe-instances --filters "Name=instance-state-name,Values=running" \
+    "Name=instance-id,Values=${INSTANCE_ID}" --query 'Reservations[*].Instances[*].[PublicIpAddress]' --output text)
+echo INSTANCE_PUBLIC_IP = ${INSTANCE_PUBLIC_IP}
+
+echo wait for instance to connect to clucter
+INSTANCE_COUNT=0
+while [ ${INSTANCE_COUNT} -eq 0 ]
+do
+    INSTANCE_COUNT=$(aws --region ${region} ecs list-container-instances --cluster $cluster_name | jq -r '.containerInstanceArns | length')
+    echo -n .
+done
+
+echo registering task definition
+TASK_REVISION=$(aws --region ${region} ecs register-task-definition --cli-input-json file://tast_def.json | jq -r '.taskDefinition | .revision')
+echo TASK_REVISION = ${TASK_REVISION}
+
+echo run task
+TASK_ARN=$(aws --region ${region} ecs run-task --count 1 --cluster $cluster_name --task-definition $task_family | jq -r '.tasks | .[0] | .taskArn')
+echo TASK_ARN = ${TASK_ARN}
+echo wait for task to be running
+aws --region ${region} ecs wait tasks-running  --cluster $cluster_name --tasks ${TASK_ARN} > /dev/null
+
+echo get page 
+ERROR_CODE=0
+PAGE=$(curl --data x= http://${INSTANCE_PUBLIC_IP} | grep "kontain.KKM")
```

Comment:
in above line with $(curl --data x=
not sure what x= refers to

Reply:
We need to get curl to do a post on this page to imitate user pressing Test button. Passing any data works so x= represents passing a dummy data.

Reply:
Got it. Sorry, thought x= was a param.  Makes sense. Thanks.